diff --color -ur a/scipy/integrate/setup.py b/scipy/integrate/setup.py
--- a/scipy/integrate/setup.py	2019-07-18 01:15:44.000000000 +0200
+++ b/scipy/integrate/setup.py	2020-10-16 07:56:48.150288830 +0200
@@ -30,8 +30,7 @@
     quadpack_test_src = [join('tests','_test_multivariate.c')]
     odeint_banded_test_src = [join('tests', 'banded5x5.f')]
 
-    config.add_library('mach', sources=mach_src,
-                       config_fc={'noopt':(__file__,1)})
+    config.add_library('mach', sources=mach_src)
     config.add_library('quadpack', sources=quadpack_src)
     config.add_library('lsoda', sources=lsoda_src)
     config.add_library('vode', sources=vode_src)
diff --color -ur a/scipy/special/setup.py b/scipy/special/setup.py
--- a/scipy/special/setup.py	2020-01-20 21:44:32.000000000 +0100
+++ b/scipy/special/setup.py	2020-10-16 07:55:35.333251058 +0200
@@ -55,8 +55,7 @@
     amos_src = [join('amos','*.f')]
     cdf_src = [join('cdflib','*.f')]
     specfun_src = [join('specfun','*.f')]
-    config.add_library('sc_mach',sources=mach_src,
-                       config_fc={'noopt':(__file__,1)})
+    config.add_library('sc_mach',sources=mach_src)
     config.add_library('sc_amos',sources=amos_src)
     config.add_library('sc_cdf',sources=cdf_src)
     config.add_library('sc_specfun',sources=specfun_src)
@@ -107,7 +106,7 @@
 
     # Cython API
     config.add_data_files('cython_special.pxd')
-    
+
     cython_special_src = ['cython_special.c', 'sf_error.c', '_logit.c.src',
                           "amos_wrappers.c", "cdf_wrappers.c", "specfun_wrappers.c"]
     cython_special_dep = (headers + ufuncs_src + ufuncs_cxx_src + amos_src
